fineract-core/src/main/java/org/apache/fineract/commands/service/CommandSourceService.java

```diff
@@ -64,8 +66,15 @@ public CommandSource saveInitialSameTransaction(CommandWrapper wrapper, JsonComm
 
     @NotNull
     private CommandSource saveInitial(CommandWrapper wrapper, JsonCommand jsonCommand, AppUser maker, String idempotencyKey) {
-        CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
-        return commandSourceRepository.saveAndFlush(initialCommandSource);
+        try {
+            CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
+            return commandSourceRepository.saveAndFlush(initialCommandSource);
+        } catch (JpaSystemException jse) {
+            if (jse.getRootCause().getMessage().contains("UNIQUE_PORTFOLIO_COMMAND_SOURCE")) {
```

Comment:
Please use null checks! Root cause and message both can be null!

Reply:
Done! null checks added